### DISCUSSION TITLE
Does this still work with DSM 7.2.2 and DS3622xs+ / Ds2422+ ?

### DISCUSSION THREAD

**Discussion by tommyd355** [007revad/Synology_HDD_db#353] [Q&A] [answered]
Saw a posting that one of the scripts no longer work with DSM 7.2.2. I use this for my DS3622xs+ with EXOS hard drives and am uncertain about upgrading because one cannot go back !!! The version I last ran was 3.1.64 though I see that 3.5.101 is out now. Thanks !!!

**Comment by 007revad** [comment]
It still works in DSM 7.2.2

**Comment by tommyd355** [comment, marked as answer]
Updated DS3622xs+ (with 11 16TB EXOS drives) to DSM 7.2.2.
Reran the script and rebooted.
Drives are all now "Healthy" !!!
Thank You !!!

**Comment by MorphyDK** [comment]
Hi @007revad 


Just got me a DS2422+ , I just have some minor questions for you.
Filled up with WD 20TB Gold drives - all drives were not verified after setup which I expected.

I opened /etc.defaults/synoinfo.conf to make a change: 

support_disk_compatiblity="yes" changed to "no"

As I see all the WD drives are now healthy :) 

But then im in doubt if I can change: 

support_m2_pool=2"no" to "yes"

Will this actually work just like your script? So im able to make my WD SN700 drives available to create a volume?

Guessing an update from Synology to DSM might change the settings in this file, so I can just enable again if the problem persist? 
Or do I need some critical changes made via your script?

Im just worried that Synology in the future might strict some more things and then im locked with your script and you know.. I might have major issues with my data  :(

**Reply by 007revad** [reply]
The DS2422+ has no internal M.2 slots, so I assume you're getting a E10M20-T1 or M2D20 so you can use NVMe drives?

Changing support_m2_pool="no" to "yes" is only part of it. The syno_hdd-db script is required to be able to create a volume on 3rd party NVMe drives in storage manager.

Officially DSM will not let you create a volume on NVMe drives in a E10M20-T1 or M2D20 (even with Synology NVMe drives). You need the syno_hdd-db script to be able to do this.

You should schedule the script to run as root at boot to set your changes after a DSM update.

**Reply by MorphyDK** [reply]
Ahhh yes sorry forgot to mention that :)

I have M2D20 with two drives on it only running as cache.

Thanks for the fast reply - I will look into the script tonight. Amazing job 🥇 

Have a nice weekend

**Comment by MorphyDK** [comment]
@007revad 

I got my NVME drive working as volume :) . 

First I ran the DB HD script , and I saw my drives were added to the DB. All healthy. 

Then I want to make the m2 pool, but I couldn't since there was no drives available?
After I ran your syno_hdd-db - I couldn't make the drives to a pool / volume unless I ran Synology enable m2 volume?

Im was bit confused of all the different scripts. 

If I want to roll back if I want to use them as cache instead of volume , how do I do it ?  

I tried sudo -s /volume1/scripts/syno_enable_m2_volume.sh -r to make a restore, and then the system prompted me that there was no backup file? 

If I manually have to restore everything, lets say if the nas needs to be returned due failure or something. 

How do I manually roll everything back to normal ? 

Appreciate your help :)

**Reply by 007revad** [reply]
Just delete the storage pool in Storage Manager and it will let you use the NVMe drives as a cache.

During a DSM update it deletes the backup files, and all changes, the script made. If DSM hasn't been updated since the script was run the -r or --restore option would restore everything.

If the NAS is still working you can reinstall DSM https://github.com/007revad/Synology_DSM_reinstall

If the NAS is not working you can connect the drives to a computer and delete the partitions.

If you are sending back a NAS without drives there's nothing on the NAS that's been changed by the script.

**Reply by MorphyDK** [reply]
Thank you so much for the explanation 😊👍.

Everything is running perfectly. 

Cheers

**Comment by bat0nas** [comment]
Could anybody confirm that the script is compatible with freshly released:

DSM 7.2.2-72806 Update 1

Thanks!!

**Reply by 007revad** [reply]
Yes
